daemon/networkdriver/portallocator/portallocator.go

```diff
@@ -59,6 +68,29 @@ func NewErrPortAlreadyAllocated(ip string, port int) ErrPortAlreadyAllocated {
 	}
 }
 
+func init() {
+	const param = "/proc/sys/net/ipv4/ip_local_port_range"
+
+	if line, err := ioutil.ReadFile(param); err != nil {
+		log.Errorf("Failed to read %s kernel parameter: %s", param, err.Error())
+	} else {
+		var start, end int
+		if n, err := fmt.Fscanf(strings.NewReader(string(line)), "%d\t%d", &start, &end); n != 2 || err != nil {
+			if err == nil {
+				err = fmt.Errorf("unexpected count of parsed numbers (%d)", n)
+			}
+			log.Errorf("Failed to parse port range from %s: %v", param, err)
+		} else {
+			beginPortRange = start
+			endPortRange = end
+		}
+	}
+}
+
+func GetPortRange() (int, int) {
```

Comment:
@LK4D4 he calls it from https://github.com/docker/docker/pull/10238/files#diff-b2058005d9cae888c97c7a0a9ce8e797R132

@miminar I would rename it to `PortRange` though.